x-pack/plugins/enterprise_search/public/applications/app_search/components/multi_input_rows/multi_input_rows.tsx

```diff
@@ -54,7 +56,15 @@ export const MultiInputRows: React.FC<Props> = ({
   }, [values]);
 
   return (
-    <>
+    <EuiForm
+      id={id}
+      component="form"
+      onSubmit={(e) => {
+        e.preventDefault();
+        if (onSubmit) onSubmit(filterEmptyValues(values));
+      }}
+      data-test-subj="multiInputRowsForm"
+    >
```

Comment:
Sorry, totally forgot to leave a comment here earlier, but this change (+ an option to hide an OOTB submit button) is to get the MultiInputForm component working with a submit button in the Synonyms modal footer. Check out the gif in the PR description as well as EUI's documentation:

https://elastic.github.io/eui/#/layout/modal#forms-in-a-modal

Reply:
My only concern with this ... is that in places like Relevance Tuning, we'll end up having nested forms.

Before I give this too much though, is this something you've already considered?

Reply:
Ahh, thanks for catching this Jason - I hadn't realized the relevance tuning parent was a form. That's a good catch, I'll have to rework the API for this component to either be a form/onSubmit or div/onChange.

Reply:
https://github.com/elastic/kibana/pull/98344/commits/fea09c770accc16e67d8e7ccfcc9f11b26dca418 - let me know if this looks good to you. I tested all 3 views (Curations, Synonyms, Relevance Tuning) and they're still working as before. I think it _should_ fix the nested `<form>` issue! 🙏

Reply:
On a higher level:

> I have been thinking of this as a form component, so something to be nested inside of a form. This component now has become a form itself, which I think could be confusing / create some weird HTML.

I think it can be either (both its own form and a form child/component). For Curations & Synonyms it's definitely a standalone form, for Relevance Tuning it's nested, so I think it makes sense to support both. Hopefully that's not too confusing!